onboarding/source/amb_light_service.c

```diff
@@ -17,37 +21,119 @@
-static void lightServiceTask(void * pvParameters);
+static void lightServiceTask(void *pvParameters);
 
-obc_error_code_t initLightService(void) {
+obc_error_code_t initLightService(void)
+{
     /* USER CODE BEGIN */
     // Create the task and queue here. Return error code if task/queue was not created successfully.
+    if (lightServiceTaskHandle == NULL)
+    {
+        lightServiceTaskHandle = xTaskCreateStatic(lightServiceTask,
+                                                   LIGHT_SERVICE_NAME,
+                                                   LIGHT_SERVICE_STACK_SIZE,
+                                                   NULL,
+                                                   LIGHT_SERVICE_PRIORITY,
+                                                   lightServiceTaskStack,
+                                                   &lightServiceTaskBuffer);
+    }
+
+    if (lightServiceTaskHandle == NULL)
+    {
+        return OBC_ERR_CODE_TASK_CREATION_FAILED;
+    }
+
+    if (queueHandle == NULL)
+    {
+        queueHandle = xQueueCreateStatic(QUEUE_LENGTH,
+                                         ITEM_SIZE,
+                                         ucQueueStorageArea,
+                                         &queueBuffer);
+    }
+
+    if (queueHandle == NULL)
+    {
+        return OBC_ERR_CODE_QUEUE_CREATION_FAILED;
+    }
 
+    return OBC_ERR_CODE_SUCCESS;
     /* USER CODE END */
 }
 
-static void lightServiceTask(void * pvParameters) {
+uint16_t getLightSensorData(void)
+{
+    adcData_t adc_data;
+
+    /** - Wait for ADC Group1 conversion to complete */
+
+    adcStartConversion(adcREG1, adcGROUP1);
+    /** - Read the conversion result
+     *     The data contains the Light sensor data
+     */
+    while (!adcIsConversionComplete(adcREG1, adcGROUP1))
+        ;
+    adcGetData(adcREG1, adcGROUP1, &adc_data);
+    return adc_data.value;
+}
+
+static void lightServiceTask(void *pvParameters)
+{
     /* USER CODE BEGIN */
     // Wait for MEASURE_LIGHT event in the queue and then print the ambient light value to the serial port.
+    if (queueHandle != NULL)
+    {
+        light_event_t lightEvent;
+        while (1)
+        {
+            // queueBuffer has different type with event, will it convert automatically?
+            if (xQueueReceive(queueHandle, &lightEvent, (TickType_t)0) == pdTRUE)
+            {
+                if (lightEvent == MEASURE_LIGHT)
+                {
+                    uint16_t lightData = getLightSensorData();
+                    // use %lu if uint32_t (long unsigned int)
+                    sciPrintf("%u\n", lightData);
+                };
+            }
+        }
+    }
 
     /* USER CODE END */
 }
 
-obc_error_code_t sendToLightServiceQueue(light_event_t *event) {
+obc_error_code_t sendToLightServiceQueue(light_event_t *event)
+{
     /* USER CODE BEGIN */
     // Send the event to the queue. Return error code if event was not sent successfully.
-    
+    if (queueHandle != NULL)
```

Comment:
If `queueHandle` is `NULL`, we should return a more appropriate error code than just `OBC_ERR_CODE_QUEUE_SEND_FAILED` or `OBC_ERR_CODE_QUEUE_FULL`. Return something like `OBC_ERR_CODE_INVALID_STATE` instead.

Reply:
Also, instead of using a nested if statement here, it's more clear to write something like:
```
if (queueHandle == NULL)
    return OBC_ERR_CODE_INVALID_STATE;
    
if (xQueueSend(queueHandle, event, (TickType_t)0) == pdTRUE)
    return OBC_ERR_CODE_SUCCESS;

return OBC_ERR_CODE_QUEUE_FULL;
```